go-controller/pkg/ovn/ovn.go

```diff
@@ -453,23 +453,25 @@ func (oc *Controller) iterateRetryPods(updateAll bool) {
 		if podEntry.ignore {
 			continue
 		}
+
 		pod := podEntry.pod
-		if !util.PodScheduled(pod) {
+		podDesc := fmt.Sprintf("[%s/%s/%s]", pod.UID, pod.Namespace, pod.Name)
+		// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry
+		kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			if e, ok := err.(*errors.StatusError); ok && e.ErrStatus.Code == http.StatusNotFound {
```

Comment:
@trozet while we're here, we should just use `IsNotFound(err)` instead of open-coding it.